Sources/SpeziFHIR/FHIRStore.swift

```diff
@@ -19,149 +17,146 @@ import Spezi
 /// The ``FHIRStore`` is automatically injected in the environment if you use the ``FHIR`` standard or can be used as a standalone module.
 @Observable
 public final class FHIRStore: Module,
-                              EnvironmentAccessible,
-                              DefaultInitializable,
-                              @unchecked Sendable /* `unchecked` `Sendable` conformance fine as access to `_resources` protected by `NSLock` */ {
-    private let lock = NSLock()
-    @ObservationIgnored private var _resources: [FHIRResource]
-    
-    
-    /// Allergy intolerances.
+                             EnvironmentAccessible,
+                             DefaultInitializable,
+                             Sendable {
+    /// Actor-isolated storage for `FHIRResource`s backing the ``FHIRStore``.
+    private actor Storage {
+        // Non-isolation required so that resource access via the `FHIRStore` stays sync (required for seamless SwiftUI access).
+        // Isolation is still guaranteed as the only modifying functions `insert()` and `remove()` are isolated on the `FHIRStore.Storage` actor.
+        nonisolated(unsafe) private var _resources: [FHIRResource] = []
```

Comment:
Synchronizing modifications to a single task but reading from other threads might still lead to race conditions as CPU caches might render outdated information (or out of order issues). The previous lock implementation would have protected against this as it makes sure that shared state is propagated to all CPU caches.
Just using a lock, however, doesn't  prevent inconsistent reads. E.g., a view might read this property twice in its view body potentially receiving two different values (e.g., a isEmpty check might read the property a second time). Either, we need to make this behavior explicit (e.g., as part of documentation or through API design).
Most importantly, `FHIRResource` is not Sendable. So this is unsafe by definition and is all hidden by the `nonisolated(unsafe)`. `FHIRResources` cannot be shared between two actors (they might be sent, however we always keep a reference through this array. So this is impossible to be provable correct by the compiler).

As it seems this can be semantically considered a view model, why not just synchronize to the MainActor? We might do initial population from the array on a different actor (if necessary?), as we can `send` the result to the array that is isolated on, e.g., MainActor.